core/src/consensus.rs

```diff
@@ -31,15 +32,107 @@ pub enum ConsensusLongRangeForkDecisionReason {
     StateHash,
 }
 
+/// Consensus timing information for a specific slot.
+///
+/// Represents the temporal context of a consensus slot within the Ouroboros Samasika
+/// protocol, including epoch boundaries and global slot numbering. This structure
+/// is essential for coordinating consensus operations across the network.
+///
+/// ## Related Specification
+///
+/// Based on the [Mina Consensus Specification](https://github.com/MinaProtocol/mina/blob/compatible/docs/specs/consensus/README.md)
+/// which defines the Ouroboros Samasika consensus mechanism.
+///
+/// ## Timing Hierarchy
+///
+/// The consensus protocol operates on a hierarchical timing structure:
+/// - **Slots**: Basic time units (3 minutes each)
```

Comment:
Note that we want to lower to 90 seconds, see https://github.com/o1-labs/openmina/issues/1259?issue=o1-labs%7Copenmina%7C1265